nighthawk/source/client/benchmark_client_impl.cc

```diff
@@ -0,0 +1,243 @@
+#include "nighthawk/source/client/benchmark_client_impl.h"
+
+#include "absl/strings/str_split.h"
+
+#include "envoy/event/dispatcher.h"
+#include "envoy/thread_local/thread_local.h"
+
+#include "extensions/transport_sockets/well_known_names.h"
+
+#include "common/common/compiler_requirements.h"
+#include "common/http/header_map_impl.h"
+#include "common/http/headers.h"
+#include "common/http/http1/conn_pool.h"
+#include "common/http/http2/conn_pool.h"
+#include "common/http/utility.h"
+#include "common/network/dns_impl.h"
+#include "common/network/raw_buffer_socket.h"
+#include "common/network/utility.h"
+#include "common/upstream/cluster_manager_impl.h"
+#include "common/upstream/upstream_impl.h"
+
+#include "nighthawk/common/statistic.h"
+
+#include "nighthawk/source/client/stream_decoder.h"
+
+using namespace std::chrono_literals;
+
+namespace Nighthawk {
+namespace Client {
+
+BenchmarkClientHttpImpl::BenchmarkClientHttpImpl(Envoy::Api::Api& api,
+                                                 Envoy::Event::Dispatcher& dispatcher,
+                                                 Envoy::Stats::StorePtr&& store,
+                                                 StatisticPtr&& connect_statistic,
+                                                 StatisticPtr&& response_statistic,
+                                                 const std::string& uri, bool use_h2)
+    : api_(api), dispatcher_(dispatcher), store_(std::move(store)),
+      scope_(store_->createScope("client.benchmark.")),
+      connect_statistic_(std::move(connect_statistic)),
+      response_statistic_(std::move(response_statistic)), use_h2_(use_h2),
+      uri_(std::make_unique<Uri>(Uri::Parse(uri))),
+      benchmark_client_stats_({ALL_BENCHMARK_CLIENT_STATS(POOL_COUNTER(*scope_))}) {
+  ASSERT(uri_->isValid());
+
+  connect_statistic_->setId("benchmark_http_client.queue_to_connect");
+  response_statistic_->setId("benchmark_http_client.request_to_response");
+
+  request_headers_.insertMethod().value(Envoy::Http::Headers::get().MethodValues.Get);
+  request_headers_.insertPath().value(uri_->path());
+  request_headers_.insertHost().value(uri_->host_and_port());
+  request_headers_.insertScheme().value(uri_->scheme() == "https"
+                                            ? Envoy::Http::Headers::get().SchemeValues.Https
+                                            : Envoy::Http::Headers::get().SchemeValues.Http);
+}
+
+bool BenchmarkClientHttpImpl::syncResolveDns() {
+  bool dns_resolved = false;
+  auto dns_resolver = dispatcher_.createDnsResolver({});
+  std::string to_resolve = uri_->host_without_port();
+
+  if (to_resolve.size() > 0 && to_resolve[0] == '[') {
```

Comment:
What is this for? Please add comment in source as well as review.

Reply:
This was checking to see if the hostname represented an ipv6 address; this code changed a bit and now avoids forwarding ip adresses to dns.